### PR TITLE
Several issues

### PR DESCRIPTION
**Pull request by ConConRob**
# Description

- show hotel name
- show staff name when joined chat
- ask for rating on reload when needed
- removes rating request if message is sent
- added spinner to login button
- added typing component
- handles checkout
- sends message on enter press

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

#24 #23 #22 #21 #14 #26 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

- [ ] This change requires a documentation update

## Change status

- [x] Complete, tested, ready to review and merge

- [ ] Complete, but not tested (may need new tests)

- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
Manually 

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [x] I have made corresponding changes to the documentation, if relevant

- [x] I have run the app using my feature and ensured that no functionality is broken

- [x] My changes generate no new warnings

- [x] There are no merge conflicts

### CODE REVIEW

File: src/requests/ajax.js [hunk 1]
```diff
@@ -3,15 +3,15 @@ import axios from 'axios';
...
+  const res = await axios.post(`${DOMAIN}${PATH.users}/chat`, {
+    name,
+    passcode,
+  });
+  return res.data; // {user, token}
+}
+
+export async function getHotel(hotel_id) {
+  const res = await axios.get(`${DOMAIN}${PATH.hotel}/${hotel_id}`);
+  return res.data;
```
Comment (Pav0l) [inline src/requests/ajax.js:16]: Why remove the `try-catch` blocks? This can cause unhandled promise rejections

File: src/views/LoginView.js [hunk 2]
```diff
@@ -39,44 +44,57 @@ class LoginView extends React.Component {
...
-      .catch(error => {
-        // display a flash message
-        console.log(error);
-      });
+    const { nameInput, passcodeInput } = this.state;
+    if (nameInput && passcodeInput) {
+      this.setState({ loggingIn: true, errorMessage: '' });
+      loginRequest(this.state.nameInput, this.state.passcodeInput)
+        .then(res => {
+          debugger;
```
Comment (Pav0l) [inline src/views/LoginView.js]: De 🐛 er

File: src/views/LoginView.js [hunk 3]
```diff
@@ -39,44 +44,57 @@ class LoginView extends React.Component {
...
+          this.props.setUserAndToken(res.user, res.token);
+        })
+        .catch(error => {
+          // display a flash message
+          debugger;
```
Comment (Pav0l) [inline src/views/LoginView.js]: De 🐛 er

**Review by Pav0l — CHANGES_REQUESTED** [review]
Debuggers and try-catch blocks

File: src/requests/ajax.js (hunk 1 above)
Comment (ConConRob) [reply src/requests/ajax.js:16]: It is handled in the component if I handle here and return it will never hit the catch in the component

**Review by Pav0l — APPROVED** [review]
Good job!

**Pull request merged**